environments/kubernetes/kube-thanos.libsonnet

```diff
@@ -38,7 +43,11 @@ local deployment = k.apps.v1.deployment;
         service.mixin.metadata.withNamespace(namespace),
       statefulSet+: {
         metadata+: {
+          name: 'thanos-receive-default',
```

Comment:
I think making store `sts.mixin.spec.withReplicas(3),` is not that helpful as you have headless service with `dnssrv` option. This means that you treat each store as separate source that will be touch 100% time. If request will bring it down all would go down. However it increase availability on other crashes / restarts (: 

We explain this more: https://github.com/improbable-eng/thanos/issues/1083